### DISCUSSION TITLE
How to change  modifier configuration when  using tree shaking?

### DISCUSSION THREAD

**Discussion by dagadbm** [floating-ui/floating-ui#1221] [Q&A] [answered]
I am not sure this should be an  issue or not but I honestly tried to  do this  but failed.

I was trying to use the tree-shaking popper version and I need to use the flip and preventOverflow modifiers, however I see no way to actually customize them (for eexample  change  the boundary/element option  from them).

Also  the tree shaking documentation is different on the README.md and  on the docs page so  that was  confusing as well.

In the end I just ended up using the normal popper.

I also don't  understand why the popper lite  still contains so many default modifications and stylings  that I don't really customize or use directly.

THanks!

**Comment by FezVrasta** [comment, marked as answer]
Hi! Do you have a code example to share? 

The example from the docs is:

```js
import { createPopper } from '@popperjs/core/lib/popper-lite';
import flip from '@popperjs/core/lib/modifiers/flip';
import preventOverflow from '@popperjs/core/lib/modifiers/preventOverflow';

createPopper(reference, popper, {
  modifiers: [flip, preventOverflow]
})
```

If you'd like to customize the modifiers, you can do the following:

```js
import { createPopper } from '@popperjs/core/lib/popper-lite';
import flip from '@popperjs/core/lib/modifiers/flip';
import preventOverflow from '@popperjs/core/lib/modifiers/preventOverflow';

createPopper(reference, popper, {
  modifiers: [
    { ...flip, options: { mainAxis: false } },
    preventOverflow
  ]
})
```

Or alternatively:


```js
import { createPopper } from '@popperjs/core/lib/popper-lite';
import flip from '@popperjs/core/lib/modifiers/flip';
import preventOverflow from '@popperjs/core/lib/modifiers/preventOverflow';

createPopper(reference, popper, {
  modifiers: [
    flip,
    preventOverflow,
    { name: 'flip', options: { mainAxis: false } },
  ]
})
```

**Reply by dagadbm** [reply]
Thanks! d'oh! Actully pretty  easy to do

Also why is the documentation on the README not pointing to the core/lib/popper-lite ?
![image](https://user-images.githubusercontent.com/16737095/105206244-0bdc0280-5b3e-11eb-97ce-339893afaae5.png)

I also would like to know  why is this the lightest weight alternative, do we really need these by  default so popper works?:
popperOffsets
computeStyles
applyStyles
eventListeners

**Reply by FezVrasta** [reply]
eventListeners, applyStyles, and computedStyles are essential to get the expected functionalities. popperOffsets is nice to have.

I think with the most recent changes you don't need to import from popper-lite anymore, the default import should work just fine with three shaking.
